src/odemis/driver/smaract.py

```diff
@@ -1254,8 +1254,18 @@ def __init__(self, name, role, locator, axes, ref_on_init=False, linear_speed=0.
         self.set_linear_speed(self.linear_speed)
         self.rotary_speed = rotary_speed
         self.set_rotary_speed(math.degrees(self.rotary_speed))
+
+        self.speed = VigilantAttribute({'x': self.linear_speed,
+                                   'y': self.linear_speed,
+                                   'z': self.linear_speed,
+                                   'rx': self.rotary_speed,
+                                   'rz': self.rotary_speed},
+                                   readonly=True)
+
+        # create a timer thread that will be used to update the position while waiting for events
+        self.update_position_timer = RepeatingTimer(0.1, self._updatePosition)
```

Comment:
No, I'd say, start with a period of 1s (ie, 1Hz), and during a move, switch to 0.05s (20Hz). You can do that by changing the RepeatingTimer.period at the beginning and end of the move (in a `finally` block).